README.md

```diff
@@ -353,10 +353,11 @@ Since this still has some overhead, we exclude some "big-name" action maintainer
 ### Updating actions guitelines
 
 To keep efforts low when updating actions, we list all forked actions here.
-To keep confusion to a minimum, the version we use is always on a `pdk-templates` branch.
-This way we can update (`git fetch`/`git push`) forked repositories with no prejudice, test out the changes, and only then update the `pdk-templates` branch.
+To keep confusion to a minimum, the version we use is always on a `pdk-templates-v1` branch.
```

Comment:
Just thinking out loud: should it follow upstream? If they are on v1, it should be v1. If they're on v2 but v1 was never deployed in pdk-templates then it's confusing when they don't match.

Reply:
Hmmm - I was thinking this is the template's version, as that's what we control here. Not sure there is one good solution.

Reply:
My reasoning is that I'd want to treat it more as a mirror than a fork so the fewer changes compared to upstream, the better. As a community member, having to rely on Puppetlabs has proven to block us from time to time. Making it easier to use upstream actions again is IMHO a good thing.

Reply:
Would that blocking concern be addressed by opening these repos up to Trusted Contributors?

Reply:
Let me try to phrase that clearer:

Would that blocking concern be addressed by giving you (and others) write access to these repositories through a Trusted Contributor Programme?

Reply:
I'm not sure. I don't use the pdk-templates so I would have no incentive to work on that.

However, I do think that you should have a rule that any modifications should be sent to the upstream action and you only merge/cherry pick from upstream. Only in exceptional areas would you want it to become a real fork. The project's README should describe that.

Reply:
> As a community member, having to rely on Puppetlabs has proven to block us from time to time. 
> I'm not sure. I don't use the pdk-templates so I would have no incentive to work on that.
I'm getting mixed messages here.

> However, I do think that you should have a rule that any modifications should be sent to the upstream action and you only merge/cherry pick from upstream. Only in exceptional areas would you want it to become a real fork. The project's README should describe that.

Agreed - there should be no changes on the `pdk-templates-*` branches that are not (at least in a PR) upstream.

Reply:
> > As a community member, having to rely on Puppetlabs has proven to block us from time to time.
> > I'm not sure. I don't use the pdk-templates so I would have no incentive to work on that.
> 
> I'm getting mixed messages here.

I'm saying that as a community member I've experienced in the past that it can be annoying to be blocked on something. In my experience, Puppet (still have to get used to not call it Puppetlabs) is slow to review PRs. Just of the top of my head, https://github.com/puppetlabs/puppetlabs_spec_helper/pull/327 has been open for a month without even a review.

Now I'm experienced in being a maintainer so I know it can be hard to deal with. What I am suggesting that the process should be designed in a way that with minimal friction users can work around it. Being as close as possible to upstream, also encourages the best open source behavior for the largest amount of people.

Does that clarify it?

Reply:
> Does that clarify it?

yes, then we're on the same page what we're talking about.

Again, would **all those concerns** be addressed by giving you (and others) write access to **all these repositories** through a Trusted Contributor Programme? Not just in this particular case, or that PR you're referencing; in the most general sense of "would the puppet community use a more cohesive toolset and move forward quicker if y'all didn't need waiting on puppet employees to push a green button for you?"

Reply:
I think in general more contributors does help. Of course there's never a guarantee, but it's the right way to deal with it IMHO.

> if y'all didn't need waiting on puppet employees to push a green button for you?"

Note that often actual releases are needed. Often you need a gem or module release for production cases. I know this can also be PRs, but I'm being explicit here.

Reply:
I should also mention that I refuse to use the PDK since it contains telemetry and fundamentally disagree with that. Having an opt-out is IMHO irrelevant. Such a thing is unacceptable to me and by using the opt-out, you would legitimate the tool.

Reply:
I've added verbiage relating to limiting modifications to the README and screencapped your comment on trusted contributor rights for product mgmt.